front/components/agent_builder/capabilities/AddToolsDropdown.tsx

```diff
@@ -75,12 +85,50 @@ export function AddToolsDropdown({
     });
   };
 
-  // TODO: Add Reasoning logic here (see how it's done in Assistant Builder).
   function onClickMCPServer(mcpServerView: MCPServerViewType) {
+    const requirement = getMCPServerRequirements(mcpServerView);
     const action = getDefaultMCPAction(mcpServerView);
-    assert(action);
 
-    if (action.noConfigurationRequired) {
+    const isReasoning = requirement.requiresReasoningConfiguration;
```

Comment:
Add missing logic to select default reasoning model, brought over from Assistant Builder